packages/web/src/components/composer-input/ComposerInput.tsx

```diff
@@ -0,0 +1,363 @@
+import { ChangeEvent, useCallback, useEffect, useRef, useState } from 'react'
```

Comment:
Are there any big changes in here to highlight from the composer component from before?

Reply:
Mostly the bullet points in the PR description. Lots of small updates to how the link detections are handled and things like that. It's basically an updated version of what was in the ChatComposer component previously